### PR TITLE
DTLS 1.3: Update protocol_version.pm

### PR DESCRIPTION
**Pull request by fwh-dc**
- [x] tests are added or updated

### CODE REVIEW

File: test/ssl-tests/protocol_version.pm [hunk 1]
```diff
@@ -336,6 +369,25 @@ sub generate_resumption_tests {
         };
     }
 
+    if (!disabled("dtls1_3") && (!disabled("ec") || !disabled("dh")) && $dtls) {
+        push @client_tests, {
+            "name" => "resumption-with-hrr",
```
Comment (mattcaswell) [inline test/ssl-tests/protocol_version.pm:374]: Don't we want a different name here?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: test/ssl-tests/protocol_version.pm (hunk 1 above)
Comment (fwh-dc) [reply test/ssl-tests/protocol_version.pm:374]: You are right. I'll fix that!

**Comment by t8m** [comment]
The CI is relevant.

**Review by t8m — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Review by shahsb — APPROVED** [review]
LGTM!

**Comment by shahsb** [comment]
IMO, at the end, when we're ready, we should squash all commits into a single one. So that in future if need arises to revert something, that really helps. Thanks!

**Comment by t8m** [comment]
@shahsb As the commits are `fixup!` commits they will be squashed automatically when merging.

**Comment by t8m** [comment]
Merged to the feature branch. Thank you for your contribution.

**Pull request closed without merging**